libraries/lib-ipc/internal/socket_guard.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <cassert>
+#include "ipc-types.h"
+
+class socket_guard
+{
+   SOCKET mSocket{INVALID_SOCKET};
+public:
+   explicit socket_guard(const socket_guard&) = delete;
+   socket_guard& operator=(const socket_guard&) = delete;
+
+   explicit socket_guard(SOCKET sock = INVALID_SOCKET) noexcept : mSocket(sock) { }
+
+   explicit socket_guard(socket_guard&& rhs) noexcept
+      : mSocket(rhs.mSocket)
+   {
+      rhs.mSocket = INVALID_SOCKET;
+   }
+
+   explicit operator bool() const noexcept
+   {
+      return mSocket != INVALID_SOCKET;
+   }
+
+   socket_guard& operator=(socket_guard&& rhs) noexcept
+   {
+      if(this == &rhs)
+         return * this;
+
+      assert(mSocket == INVALID_SOCKET || rhs.mSocket != mSocket);
```

Comment:
This asserts “the socket was uniquely guarded”, very informally.

I would make that a `@pre` comment.

Reply:
Less urgent while this remains “internal”